ci: Fix pipeline tests

spydon opened the pull request:
Closes #34

spydon commented on packages/spec_cli/test/spec_cli_test.dart:
```diff
@@ -978,30 +989,71 @@ void main() {
-''');
+Time:        00:00:00''';
+
+        if (lastFrame.endsWith('''
+$failingTest
+$testResults
+''')) {
+          expect(
+            lastFrame,
+            endsWith(
+              '''
+$groupFailingTest
+$failingTest
+$testResults
+''',
+            ),
+          );
+        } else {
+          expect(
+            lastFrame,
+            endsWith(
+              '''
+$failingTest
+$groupFailingTest
+$testResults
+''',
+            ),
+          );
+        }
```
Is it possible to write this in a nicer way? Is it possible to do some "either or"-matcher or something?

trejdych replied:
https://api.flutter.dev/flutter/package-matcher_matcher/anyOf.html ?

spydon replied:
I searched so much for this yesterday because I knew it existed, thanks!

rrousselGit commented on packages/spec_cli/test/spec_cli_test.dart:
```diff
@@ -171,16 +171,17 @@ void main() {
 
+        final lastFrame = testRenderer.frames.last;
+        expect(lastFrame, contains('PASS  ../b/test/another_test.dart'));
+        expect(lastFrame, contains('PASS  test/my_test.dart'));
```
I'm not super fond of this approach
It doesn't test the absence of anything extra

For example the output could be:

```dart
 PASS  ../b/test/another_test.dart
 PASS  test/my_test.dart
 PASS  test/my_test.dart
 PASS  test/my_test.dart
 PASS  test/my_test.dart

Test Suites: 2 passed, 2 total
Tests:       2 passed, 2 total
Time:        00:00:00
```

which isn't good

spydon replied:
If we do something permutations of `anyOf` then, where we include all the strings?

spydon replied:
Or maybe even simpler, that we verify that the number of lines in the output also is correct?

spydon replied:
Alright, go ahead! But maybe keep this PR as a backup then, so that the pipeline isn't broken for yet another year? :sweat_smile:

rrousselGit commented:
Hum my solution didn't work. LEt's merge this then

rrousselGit commented:
Thanks for your work!

spydon commented:
> Thanks for your work!

Thanks! Could you maybe put the `hacktoberfest-accepted` tag on it too (trying to get some Hacktoberfest PRs outside of our own organization in)? 😄

> Hum my solution didn't work. LEt's merge this then

I updated one of the tests to check the number or lines and the number or empty lines, to make sure nothing unsuspected could be printed, should I do this for the rest of the tests too? I think only the tests that I updated in this PR are affected.

rrousselGit commented:
The repo is tagged with hacktoberfest already. Labelling the pr shouldn't be needed

rrousselGit commented:
> I updated one of the tests to check the number or lines and the number or empty lines, to make sure nothing unsuspected could be printed, should I do this for the rest of the tests too? I think only the tests that I updated in this PR are affected.

 I still don't really like this solution
What I would like is to have a way to force the test order. 

The tricky part is having a solution that works between different `main`

spydon commented:
>  I still don't really like this solution
> What I would like is to have a way to force the test order. 
> 
> The tricky part is having a solution that works between different `main` 

If we force a test result order, it doesn't feel like the test is actually properly testing the real output?

rrousselGit commented:
Fair enough, let me mark the PR

rrousselGit commented:
> If we force a test result order, it doesn't feel like the test is actually properly testing the real output?

It is. It's just that the output isn't random anymore, it's deterministic

spydon commented:
> It is. It's just that the output isn't random anymore, it's deterministic

Isn't that part of what these tests are testing? That even though there are races in the output, the final output will be properly formatted. Because it should not be modified to run deterministically when it is running a user's test suite right, only in the CI?

rrousselGit commented:
No. That's like if I were to test the rendering of a clock. I don't want my test to run using the CI's time. I want to mock the time to test a very specific scenario.

spydon commented:
> No. That's like if I were to test the rendering of a clock. I don't want my test to run using the CI's time. I want to mock the time to test a very specific scenario.

That makes sense, thanks for clarifying!
